### PR TITLE
Support to set different configs for different clients

### PR DESCRIPTION
**Pull request by DavdGao**
as the title says.

### CODE REVIEW

File: federatedscope/main.py [hunk 1]
```diff
@@ -34,8 +35,13 @@
 
     init_cfg.freeze()
 
+    # allow different settings for different clients
+    # cfg_client.merge_from_file(args.cfg_client)
+    cfg_client = CfgNode.load_cfg(open(args.cfg_client, 'r'))
```
Comment (xieyxclack) [inline federatedscope/main.py:40]: We should consider the situation that `args.cfg_client=None` or invalid path.

File: federatedscope/core/fed_runner.py [hunk 2]
```diff
@@ -208,6 +210,12 @@ def _setup_client(self, client_id=-1, client_model=None):
 
         if self.client_class:
             client_specific_config = self.cfg.clone()
+            # Modify the setting according to self.cfg_client in standalone mode
+            if self.cfg_client is not None:
+                client_specific_config.defrost()
+                client_specific_config.merge_from_other_cfg(self.cfg_client.get('client_{}'.format(client_id)))
+                client_specific_config.freeze()
```
Comment (yxdyc) [inline federatedscope/core/fed_runner.py:217]: The `freeze()` will override the cfg.yaml file. We may need to add a parameter for this function to skip the save process

Comment (DavdGao) [reply federatedscope/core/fed_runner.py]: It is modifed accordingly.

File: federatedscope/main.py (hunk 1 above)
Comment (DavdGao) [reply federatedscope/main.py]: It is modifed accordingly

File: federatedscope/core/configs/config.py [hunk 3]
```diff
@@ -48,16 +48,18 @@ def clean_unused_sub_cfgs(self):
...
-            with redirect_stdout(outfile):
-                tmp_cfg = copy.deepcopy(self)
-                tmp_cfg.cfg_check_funcs = []
-                print(tmp_cfg.dump())
+
+        if save:
+            # save the final cfg
+            with open(os.path.join(self.outdir, "config.yaml"), 'w') as outfile:
+                from contextlib import redirect_stdout
+                with redirect_stdout(outfile):
```
Comment (joneswong) [inline federatedscope/core/configs/config.py:59]: i am curious about why writing to a file by redirecting stdout.

**Review by joneswong — APPROVED** [review]
LGTM

**Pull request merged**